Mathlib/Combinatorics/SimpleGraph/Metric.lean

```diff
@@ -43,105 +41,176 @@ variable {V : Type*} (G : SimpleGraph V)
 
 /-! ## Metric -/
 
+section edist
+
+/--
+The extended distance between two vertices is the length of the shortest walk between them.
+It is `⊤` if no such walk exists.
+-/
+noncomputable def edist (u v : V) : ℕ∞ :=
+  ⨅ w : G.Walk u v, w.length
+
+variable {G} {u v w : V}
+
+theorem edist_eq_sInf : G.edist u v = sInf (Set.range fun w : G.Walk u v ↦ (w.length : ℕ∞)) := rfl
+
+protected theorem Reachable.exists_walk_length_eq_edist (hr : G.Reachable u v) :
+    ∃ p : G.Walk u v, p.length = G.edist u v :=
+  csInf_mem <| Set.range_nonempty_iff_nonempty.mpr hr
+
+protected theorem Connected.exists_walk_length_eq_edist  (hconn : G.Connected) (u v : V) :
+    ∃ p : G.Walk u v, p.length = G.edist u v :=
+  (hconn u v).exists_walk_length_eq_edist
+
+theorem edist_le (p : G.Walk u v) :
+    G.edist u v ≤ p.length :=
+  sInf_le ⟨p, rfl⟩
+
+@[simp]
+theorem edist_eq_zero_iff :
+    G.edist u v = 0 ↔ u = v := by
+  apply Iff.intro <;> simp [edist, ENat.iInf_eq_zero]
+
+theorem edist_self : edist G v v = 0 :=
+  edist_eq_zero_iff.mpr rfl
+
+theorem pos_edist_of_ne (hne : u ≠ v) :
```

Comment:
alias `Ne.edist_pos` maybe?

also I think this should be edist_pos_of_ne

Reply:
Should I also change the `dist` versions of the lemmas and alias them too?

Reply:
Are you sure about this alias Bhavik? This is putting a graph-specific theorem in the global Ne namespace. We could have `Ne.simpleGraph_walk_edist_pos` or something like that, but otherwise I'd suggest not having such an alias.

Reply:
Oh true, `Ne.edist_pos` is probably too non-specific to be global